### PR TITLE
executor: fix a logical error which is introduced in a previous refactor

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Fix https://github.com/pingcap/tidb/issues/11059

### What is changed and how it works?

In the refactor of PR 11018, an if-else logic is reversed

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @jackysp @suzaku

File: executor/write_test.go [hunk 1]
```diff
@@ -2439,3 +2442,12 @@ func (s *testSuite4) TestAutoIDInRetry(c *C) {
 	tk.MustExec("insert into t values ()")
 	tk.MustQuery(`select * from t`).Check(testkit.Rows("1", "2", "3", "4", "5"))
 }
+
+func (s *testSuite4) TestIssue11059(c *C) {
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create table t (pk int primary key, uk int unique, v int)")
+	tk.MustExec("insert into t values (2,11,215)")
```
Comment (jackysp) [inline executor/write_test.go]: Plz keep aligning.

**Comment by tiancaiamao** [comment]
Data race fixed in https://github.com/pingcap/tidb/pull/11063
/run-unit-test

File: executor/write_test.go (hunk 1 above)
Comment (tiancaiamao) [reply executor/write_test.go]: Done

**Comment by tiancaiamao** [comment]
CI is rather unstable currently, not caused by this PR

**Comment by lysu** [comment]
LGTM

**Pull request merged**